lib/dialects/mssql/blocks.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+var _ = require('underscore');
+
+module.exports = function(dialect) {
+	dialect.blocks.set('limit', function(params) {
+		return (!isNaN(params.offset)) ? '' : 'top(' + dialect.builder._pushValue(params.limit) + ')';
```

Comment:
MSSQL has no **LIMIT/OFFSET** Statement like mysql.
There are 2 diffrent ways:
**SELECT TOP X Fields...** is the same as **LIMIT**.
But you cannot use **OFFSET** with **TOP**. IN MSSQL 2012 and higher a new statement was added.
**SELECT fields from ... ORDER BY A OFFSET Y ROWS FETCH NEXT X ROWS ONLY**

That is why he checks for the params.offset. If it is used the top() statement will be removed and the offset/fetch statement will include the limit.